framework/system/ifsoperations.h

```diff
@@ -46,6 +46,13 @@ class IFsOperations : MODULE_EXPORT_INTERFACE
 
     virtual RetVal<QStringList> directoryFileList(const QString& path, const QStringList& nameFilters,
                                                   QDir::Filters filters = QDir::NoFilter) const = 0;
+
+    enum class ScanMode {
+        CurrentDirOnly,
+        IncludeSubdirs
+    };
+
+    virtual QStringList scanForFiles(const QString& dirPath, const QStringList& filesFilters, ScanMode mode) const = 0;
```

Comment:
Could you explain what is the purpose of this method? And what is the difference between it and directoryFileList?

Reply:
directoryFileList can return directories also. Plus, this method can scan directories for files recursively. Perhaps better to rename directoryFileList to entryList

Reply:
Why `scanForFiles`, not just `scanFiles`?

Reply:
Let's decide:
One method like: 
`RetVal<QStringList> scanEntry(const QString& path, const QStringList& nameFilters, EntryType type, ScanMode mode)`

where:
EntryType - Dir, Files, All
ScanMode - CurrentDirOnly, IncludeSubdirs

Multiple methods like:
`RetVal<QStringList> scanDirs(const QString& path, const QStringList& nameFilters, ScanMode mode)`
`RetVal<QStringList> scanFiles(const QString& path, const QStringList& nameFilters, ScanMode mode)`

Reply:
I have removed directoryFileList and replaced it on scanFiles